netdata.spec.in

```diff
@@ -101,13 +108,25 @@ install -m755 system/netdata-init-d \
         -s /sbin/nologin -r -d %{contentdir} netdata 2> /dev/null || :
 
 %post
+%if 0%{?suse_version}
+%service_add_post netdata.service
+%else
 %systemd_post netdata.service
+%endif
 
 %preun
+%if 0%{?suse_version}
+%service_del_preun netdata.service
+%else
 %systemd_preun netdata.service
+%endif
 
 %postun
+%if 0%{?suse_version}
+%service_del_preun netdata.service
+%else
 %systemd_postun_with_restart netdata.service
+%endif
```

Comment:
if we can also define some macro at prologue to remove these conditionals then we will have simpler sequence.